### DISCUSSION TITLE
"How do you prevent sensitive data exposure when using third-party libraries?"

### DISCUSSION THREAD

**Discussion by peace9056** [community/community#139977] [Code Security]
### Select Topic Area

Question

### Body

Hi all,

What strategies do you use to ensure sensitive data is not accidentally exposed when integrating third-party libraries into your codebase?

Looking forward to your insights!

**Comment by TiepBK9056** [comment]
To prevent sensitive data exposure when using third-party libraries:

1. Audit Libraries: Regularly review the libraries’ codebase and updates for potential vulnerabilities.
2. Use Environment Variables: Store sensitive data like API keys in environment variables, not directly in the code.
3. imit Data Exposure: Only pass minimal and necessary data to third-party libraries, avoiding sensitive information whenever possible.
4. Security Scanning: Implement automated tools like Snyk or Dependabot to detect and fix vulnerabilities in third-party libraries.

**Reply by peace9056** [reply]
Thanks a lot

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/139972#discussioncomment-10798827) for reference to why this discussion was closed and locked. Thanks!
